src/System.Private.CoreLib/shared/System/Threading/ExecutionContext.cs

```diff
@@ -51,6 +52,37 @@ public void GetObjectData(SerializationInfo info, StreamingContext context)
             throw new PlatformNotSupportedException();
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static void ResetToDefaults()
+        {
+            Thread currentThread = Thread.CurrentThread;
+            if (currentThread.ExecutionContext != null || currentThread.SynchronizationContext != null)
```

Comment:
These checks are now being done for all work items, regardless of whether they correctly reset or not?  Assuming everything is done correctly, then, this would be pure overhead in some cases, right?  What's the worst-case throughput impact in that case?

Reply:
The `null` checks should be fairly nominal and they should always be `null` (as `ExecutionContext.Run` will reset it back)?

However, I think `Thread.CurrentThread` isn't a low cost call...

Reply:
> However, I think Thread.CurrentThread isn't a low cost call...

Right, that's the main thing I was referring to.